src/common/AdaptiveTree.h

```diff
@@ -454,7 +454,8 @@ class ATTRIBUTE_HIDDEN AdaptiveTree
   AdaptiveTree();
   virtual ~AdaptiveTree();
 
-  virtual bool open(const std::string &url, const std::string &manifestUpdateParam) = 0;
+  virtual bool open(const std::string& url, const std::string& manifestUpdateParam) = 0;
+  virtual bool open(const std::string& url, const std::string& manifestUpdateParam, std::map<std::string, std::string> additionalHeaders) = 0;
```

Comment:
Just a quick question - would the additionalHeaders ever be required for a manifest update: https://github.com/xbmc/inputstream.adaptive/blob/a675541a5c10e0ff4e0ec2fa7b4c9d8306bcffb8/src/parser/DASHTree.cpp#L1669

Reply:
for my use case no, but perhaps it is better add it?

Reply:
Looking at it now I can't think of a reason we would need to redo this mid stream - I don't think I've seen more than one session open at a time, let alone changing sessions mid stream. Leave for now and if it comes up in the future it's not a big change.